vividus-extension-selenium/src/main/java/org/vividus/steps/ui/ScreenshotTakingSteps.java

```diff
@@ -19,23 +19,35 @@
 import java.io.IOException;
 import java.nio.file.Path;
 
-import javax.inject.Inject;
+import com.google.common.eventbus.EventBus;
 
 import org.jbehave.core.annotations.When;
+import org.vividus.reporter.event.AttachmentPublishEvent;
+import org.vividus.reporter.model.Attachment;
 import org.vividus.selenium.screenshot.ScreenshotTaker;
 
 public class ScreenshotTakingSteps
 {
-    @Inject private ScreenshotTaker screenshotTaker;
+    private final ScreenshotTaker screenshotTaker;
+    private final EventBus eventBus;
+
+    public ScreenshotTakingSteps(ScreenshotTaker screenshotTaker, EventBus eventBus)
+    {
+        this.screenshotTaker = screenshotTaker;
+        this.eventBus = eventBus;
+    }
 
     /**
-     * Takes a screenshot and saves it to the default location
-     * @throws IOException If an input or output exception occurred
+     * Takes a screenshot and publish it to the report
      */
     @When("I take screenshot")
-    public void whenITakeScreenshot() throws IOException
+    public void whenITakeScreenshot()
```

Comment:
```suggestion
    public void takeScreenshot()
```
I know it's a bit out of scope, but it's a good chance to change this ugly name